### PR TITLE
Support Hermetic java builds

### PR DESCRIPTION
**Pull request by pierDipi**
To support hermetic builds for Java, we will follow the POC here: https://github.com/openshift-knative/eventing-kafka-broker/pull/1273.

While this currently doesn't pass the enterprise contract, I've validated that it will in the future with Konflux team here: https://redhat-internal.slack.com/archives/C04PZ7H0VA8/p1727247236771179 with the addition of `KONFLUX-298`.

Depends on https://github.com/openshift-knative/eventing-kafka-broker/pull/1287

### CODE REVIEW

File: pkg/konfluxgen/kustomize/kustomize-java-docker-build/kustomization.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      kind: Pipeline
+replacements:
+  - source:
+      kind: Pipeline
+      fieldPath: spec.tasks.[name=build-container].taskRef.params.[name=bundle].value
+    targets:
+      - select:
+          kind: Pipeline
+        fieldPaths:
+          - spec.tasks.[name=build-container-deps].taskRef.params.[name=bundle].value
```
Comment (creydr) [inline pkg/konfluxgen/kustomize/kustomize-java-docker-build/kustomization.yaml:33]: nit: maybe add a quick comment what you do here, so somebody does not need to follow the paths in the yaml :shrug:

**Review by creydr — COMMENTED** [review]
Awesome.
Just one comment, but feel free to skip it.

/lgtm

/hold as we need to wait for https://github.com/openshift-knative/eventing-kafka-broker/pull/1287

File: pkg/konfluxgen/kustomize/kustomize-java-docker-build/kustomization.yaml (hunk 1 above)
Comment (pierDipi) [reply pkg/konfluxgen/kustomize/kustomize-java-docker-build/kustomization.yaml:33]: I will add it as a follow up!

**Pull request merged**